examples/advanced/main.go

```diff
@@ -26,29 +26,40 @@ import (
 	_ "github.com/wasilibs/nottinygc"
 
 	"sigs.k8s.io/kube-scheduler-wasm-extension/examples/advanced/plugin"
-	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/config"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/api"
 	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/enqueue"
-	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/klog"
+	handleapi "sigs.k8s.io/kube-scheduler-wasm-extension/guest/handle/api"
+	klog "sigs.k8s.io/kube-scheduler-wasm-extension/guest/klog/api"
 	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/prescore"
 	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/score"
 )
 
 // main is compiled to an exported Wasm function named "_start", called by the
 // Wasm scheduler plugin during initialization.
 func main() {
+	enqueue.SetPlugin(func(klog klog.Klog, jsonConfig []byte, h handleapi.Handle) api.EnqueueExtensions {
+		p := pluginInitializer(klog, jsonConfig, h)
+		return p.(api.EnqueueExtensions)
+	})
+	prescore.SetPlugin(func(klog klog.Klog, jsonConfig []byte, h handleapi.Handle) api.PreScorePlugin {
+		p := pluginInitializer(klog, jsonConfig, h)
+		return p.(api.PreScorePlugin)
+	})
+	score.SetPlugin(func(klog klog.Klog, jsonConfig []byte, h handleapi.Handle) api.ScorePlugin {
+		p := pluginInitializer(klog, jsonConfig, h)
+		return p.(api.ScorePlugin)
+	})
+}
+
+func pluginInitializer(klog klog.Klog, jsonConfig []byte, h handleapi.Handle) api.Plugin {
 	// The plugin package uses only normal Go code, which allows it to be
 	// unit testable via `tinygo test -target=wasi` as well normal `go test`.
 	//
-	// The real implementations, such as `config.Get()` use Wasm host functions
+	// The real implementations use Wasm host functions
 	// (go:wasmimport), which cannot be tested with `tinygo test -target=wasi`.
-	plugin, err := plugin.New(klog.Get(), config.Get())
+	plugin, err := plugin.New(klog, jsonConfig, h)
 	if err != nil {
 		panic(err)
 	}
-	// Instead of using `plugin.Set`, this configures only the interfaces
-	// implemented by the plugin. The Wasm host only calls functions imported,
-	// so this prevents additional overhead.
-	enqueue.SetPlugin(plugin)
-	prescore.SetPlugin(plugin)
-	score.SetPlugin(plugin)
+	return plugin
 }
```

Comment:
Ah, so we still need `pluginInitializer` because `plugin.New` returns error. I guess we can change `SetPlugin` to accept the function `func (klog klog.Klog, jsonConfig []byte, h handleapi.Handle) (api.Plugin, error)`?

Reply:
> And then we can just pass plugin.New to SetPlugin.

I didn't notice this point.. Thanks, fixed it